### ISSUE TITLE
MA0015 is reported in static local functions for containing method parameters

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#566]
### Version of the Meziantou.Analyzer NuGet package

2.0.68

### Rule Identifier

MA0015

### Target Framework

net8.0

### C# Language version

C# 12

### Description

For instance local functions rule works perfect

### Reproduction Steps

Minimal code:

```c#
public sealed class Class
{
    public void Foo(int i)
    {
        Local();

        static void Local()
        {
            throw new ArgumentException("Hello", nameof(i)); // <- S3928 is reported here
        }
    }
}
```

**Comment by meziantou** [comment]
I understand there are valid use-cases, but `i` is only valid in `nameof()`. As the local function is static, the `i` parameter is not accessible in other context (read or write value). So, should `i` be considered a valid parameter in the `Local` function? What about static lambda?

````c#
void Test(string test)
{
    _ = new System.Action<int>(static delegate (int a)
    {
        // Is it allowed?
        // Maybe the lambda is not invoked in the same context, so the parameter name doesn't make sense
        throw new System.ArgumentOutOfRangeException(paramName: "test");
    });
}    
````
